contracts/modules/cover/Quotation.sol

```diff
@@ -69,44 +69,73 @@ contract Quotation is Iupgradable {
-   * sum assured of all areas in which the quotation lies
-   * Unlocks the CN tokens of the cover. Updates the Total Sum Assured value.
-   * @param _cid Cover Id.
+   * @dev Expires a cover after a set period of time and changes the status of the cover
+   * @dev Reduces the total and contract sum assured
+   * @param coverId Cover Id.
    */
-  function expireCover(uint _cid) public {
-    require(checkCoverExpired(_cid) && qd.getCoverStatusNo(_cid) != uint(QuotationData.CoverStatus.CoverExpired));
+  function expireCover(uint coverId) external {
 
-    tf.unlockCN(_cid);
-    bytes4 curr;
-    address scAddress;
-    uint sumAssured;
-    (,, scAddress, curr, sumAssured,) = qd.getCoverDetailsByCoverID1(_cid);
-    if (qd.getCoverStatusNo(_cid) != uint(QuotationData.CoverStatus.ClaimAccepted))
-      _removeSAFromCSA(_cid, sumAssured);
-    qd.changeCoverStatusNo(_cid, uint8(QuotationData.CoverStatus.CoverExpired));
+    uint expirationDate = qd.getValidityOfCover(coverId);
+    require(expirationDate < now, "Quotation: cover is not due to expire");
+
+    uint coverStatus = qd.getCoverStatusNo(coverId);
+    require(coverStatus != uint(QuotationData.CoverStatus.CoverExpired), "Quotation: cover already expired");
+
+    (/* claim count */, bool hasOpenClaim, /* accepted */) = tc.coverInfo(coverId);
+    require(!hasOpenClaim, "Quotation: cover has an open claim");
+
+    if (coverStatus != uint(QuotationData.CoverStatus.ClaimAccepted)) {
+      (,, address contractAddress, bytes4 currency, uint amount,) = qd.getCoverDetailsByCoverID1(coverId);
+      qd.subFromTotalSumAssured(currency, amount);
+      qd.subFromTotalSumAssuredSC(contractAddress, currency, amount);
+    }
+
+    qd.changeCoverStatusNo(coverId, uint8(QuotationData.CoverStatus.CoverExpired));
   }
 
-  /**
-   * @dev Checks if a cover should get expired/closed or not.
-   * @param _cid Cover Index.
-   * @return expire true if the Cover's time has expired, false otherwise.
-   */
-  function checkCoverExpired(uint _cid) public view returns (bool expire) {
+  function withdrawCoverNote(address coverOwner, uint[] calldata coverIds, uint[] calldata reasonIndexes) external {
+
+    uint gracePeriod = tc.claimSubmissionGracePeriod();
 
-    expire = qd.getValidityOfCover(_cid) < uint64(now);
+    for (uint i = 0; i < coverIds.length; i++) {
+      uint expirationDate = qd.getValidityOfCover(coverIds[i]);
+      require(expirationDate.add(gracePeriod) < now, "Quotation: cannot withdraw before grace period expiration");
+    }
 
+    tc.withdrawCoverNote(coverOwner, coverIds, reasonIndexes);
   }
 
-  /**
-   * @dev Updates the Sum Assured Amount of all the quotation.
-   * @param _cid Cover id
-   * @param _amount that will get subtracted Current Sum Assured
-   * amount that comes under a quotation.
-   */
-  function removeSAFromCSA(uint _cid, uint _amount) public onlyInternal {
-    _removeSAFromCSA(_cid, _amount);
+  function getWithdrawableCoverNoteCoverIds(
```

Comment:
there's no test for this.  also why was it created?

Reply:
Needed for the UI.